### PR TITLE
Fix #4605: new keras layers

### LINKED ISSUE #4605: ML new Keras layers
Several new layer types are needed required for image processing. The initial new set could be:
- Add
- AveragePooling2D
- BatchNormalization
- Concatenate
- Conv2D
- Conv2DTranspose
- GaussianNoise
- GlobalAveragePooling2D
- MaxPooling2D
- SeparableConv2D
- UpSampling2D
- ZeroPadding2D

Some layers require more than the current 3 values shown in the UI. For example, the Conv2D may require many arguments:
Layer: Conv2D, Dimensionality: 32, Kernel Size: 3, Strides: 2, Padding: same, Activation: relu, Kernel Initializer: he_normal

For convolutional models, we may want to use the full browser width for the model configuration panel.

### PR DESCRIPTION
**Pull request by gurhar1133**
* New layers added
* Schema changed (subclass structure addded) to better support more types of layers and simplify keras layer generation
* UI code updated to work with new schema

### CODE REVIEW

File: sirepo/package_data/static/json/ml-schema.json [hunk 1]
```diff
@@ -319,13 +329,72 @@
...
-            "gaussianDropoutRate": ["Rate", "Float", 0.5],
-            "gaussianNoiseStddev": ["Standard Deviation", "Float", 0.01]
+            "dimensionality": ["Dimensionality", "Integer", 10],
+            "activation": ["Activation", "NeuralNetActivation", "relu"]
+
+        },
+        "dense": {
+            "_super": ["-", "model", "neuralNetLayer"]
+        },
+        "conv2D": {
```
Comment (mkeilman) [inline sirepo/package_data/static/json/ml-schema.json:339]: Put in alphabetical order

**Review by mkeilman — COMMENTED** [review]
Minor style stuff

**Comment by moellep** [comment]
Can we hide the layers which won't work with the current data? Something like add a flag which can be turned on during development, but doesn't show the new layers when running on alpha - prod. Then when the image data support is in place, we could enable it. It could be a hard-coded flag in the UI for now.

File: sirepo/package_data/static/js/ml.js [hunk 2]
```diff
@@ -1573,19 +1576,23 @@ SIREPO.app.directive('neuralNetLayersForm', function(appState, panelState) {
...
-                        if (field.toLowerCase().indexOf(name.toLowerCase()) == 0) {
-                            cols.push({
-                                field: field,
-                                label: layerSchema[field][0],
-                            });
-                        }
-                    });
+                    const layerSchema = SIREPO.APP_SCHEMA.model[stringsService.lcfirst(name)];
+                    if (layerSchema) {
+                        Object.keys(layerSchema).sort().reverse().forEach(function(field) {
```
[the remark this answers is not in the document]
Comment (gurhar1133) [reply sirepo/package_data/static/js/ml.js:1581]: Good call on the filter 👍

**Comment by gurhar1133** [comment]
@moellep made requested changes
